### PR TITLE
Cleaning up logging

### PR DESCRIPTION
**Pull request by HaoK**
Finish cleaning up logging for user manager/role manager

Fixes https://github.com/aspnet/Identity/issues/414

cc @divega

### CODE REVIEW

File: src/Microsoft.AspNet.Identity/UserManager.cs [hunk 1]
```diff
@@ -2183,7 +2052,12 @@ private static string GetChangeEmailPurpose(string newEmail)
                     errors.AddRange(result.Errors);
                 }
             }
-            return errors.Count > 0 ? IdentityResult.Failed(errors.ToArray()) : IdentityResult.Success;
+            if (errors.Count > 0)
+            {
+                Logger.LogWarning($"User {await GetUserIdAsync(user)} validation failed: {string.Join(";", errors.Select(e => e.Code))}.");
```
Comment (divega) [inline src/Microsoft.AspNet.Identity/UserManager.cs]: Starting to love string interpolation :smile:

**Comment by divega** [comment]
:shipit: 

Really nice cleanup! Hope we can keep the string interpolation usage.

File: src/Microsoft.AspNet.Identity/UserManager.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNet.Identity/UserManager.cs]: Based on previous changes, we do _not_ want to do any formatting up front. This passes an opaque string into the logging system and some of the logging providers can do useful things with the raw data. So, use the logging-specific format strings and pass the values in.

(@Eilon you wanted to be mentioned)

Comment (Eilon) [inline src/Microsoft.AspNet.Identity/UserManager.cs]: Correct. Logs need to use our own custom string formatting to be useful.

Comment (HaoK) [inline src/Microsoft.AspNet.Identity/UserManager.cs]: Updated to use logging params overloads everywhere

**Comment by HaoK** [comment]
e80853a4cf665d73e7e5cdcb652d3d51cd456068

**Pull request closed without merging**